Source/RuntimeImageLoader/Private/Texture2DAnimation/AnimatedTexture2D.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright 2023 Peter Leontev and Muhammad Ahmed Saleem. All Rights Reserved.
+
+#include "Texture2DAnimation/AnimatedTexture2D.h"
+#include "AnimatedTextureResource.h"
+
+float UAnimatedTexture2D::GetSurfaceWidth() const
+{
+	if (Decoder) return Decoder->GetWidth();
+	return 1.0f;
+}
+
+float UAnimatedTexture2D::GetSurfaceHeight() const
+{
+	if (Decoder) return Decoder->GetHeight();
+	return 1.0f;
+}
+
+FTextureResource* UAnimatedTexture2D::CreateResource()
+{
+	// create RHI resource object
+	FTextureResource* NewResource = new FAnimatedTextureResource(this);
+	return NewResource;
+}
+
+void UAnimatedTexture2D::Tick(float DeltaTime)
+{
+	if (!Decoder) return;
+
+	FrameTime += DeltaTime * PlayRate;
```

Comment:
You are NOT using `IsPlaying` variable in Tick. I thought we agreed that we should use every class variable correctly?

Reply:
Code Update Resolved